### DISCUSSION TITLE
Build problem : jobs::dispatch

### DISCUSSION THREAD

**Discussion by LoiX07** [pelicanmapping/osgearth#2559] [Support Q&A] [answered]
Hi all,
I did not manage to compile osgEarth on linux (Arch). I got this error : 
`
osgearth/src/osgEarth/ElevationPool.cpp:1126:26: erreur: pas de fonction correspondant à l'appel « dispatch(osgEarth::AsyncElevationSampler::getSample(const osgEarth::GeoPoint&, const osgEarth::Distance&)::<lambda(osgEarth::Cancelable&)>&, jobs::context&) »
 1126 |     return jobs::dispatch(task, c);
      |            ~~~~~~~~~~~~~~^~~~~~~~~
`

and also this one : 
`osgearth/src/osgEarth/weejobs.h:227:15: erreur: pas de type nommé « type » dans « struct std::invoke_result<osgEarth::AsyncElevationSampler::getSample(const osgEarth::GeoPoint&, const osgEarth::Distance&)::<lambda(osgEarth::Cancelable&)>(jobs::cancelable&)> »
  227 |         using result_of_t = typename std::invoke_result<F, Args...>::type;
`

Do you have any idea of what can cause this ?

Also, I'm noob with C++ so it may be a very simple issue !

Thanks in advance !

**Comment by LoiX07** [comment]
To precise, I got 3 objets I cannot build :
-ElevationPool.cpp.o
-ImageLayer.cpp.o
-TDTiles.cpp.o

**Comment by gwaldron** [comment, marked as answer]
Thanks; hopefully this is fixed now in d35a812

**Comment by LoiX07** [comment]
Thank you very much ! Great job btw !
